sklearn/utils/_encode.py

```diff
@@ -118,26 +206,66 @@ def _check_unknown(values, known_values, return_mask=False):
         Additionally returned if ``return_mask=True``.
 
     """
-    if values.dtype == object:
+    valid_mask = None
+
+    if values.dtype.kind in 'UO':
+        values_set = set(values)
+        values_set, nan_in_values = _extract_missing(values_set,
+                                                     none_is_missing=False)
```

Comment:
I'm a bit confused about why we use none_is_missing=False.

Reply:
Set difference works by default with `None`. If `None` was included as missing, we would need a little bit more logic to handle `None`. Currently `nan_in_*` can only include `nan`, which makes it slightly easier to figure out if `nan` is in the diff.

Maybe there is a way to simplify this logic a little...

Reply:
I removed this `none_is_missing` and changed the signature of `_extract_missing` to return a `MissingValues` tuple, which better holds the missing value info.